### PR TITLE
SwathWizard progress

### PR DESCRIPTION
**Pull request by cbielow**
other round of SwathWizard...

- can run OpenSwathWorkflow now
- supports drag'n'drop of input files
- some other minor fixes here and there

### CODE REVIEW

File: src/openms/source/APPLICATIONS/TOPPBase.cpp [hunk 1]
```diff
@@ -1596,21 +1596,10 @@ namespace OpenMS
       writeLog_("Exit code: " + String(qp.exitCode()));
       if (any_failure)
       {
-        qp.close();
         return EXTERNAL_PROGRAM_ERROR;
       }
     }
 
-    if (debug_level_ >= 10)
```
Comment (timosachsenberg) [inline src/openms/source/APPLICATIONS/TOPPBase.cpp:1604]: would it make sense to put it into the any failure block?

File: src/openms/source/DATASTRUCTURES/Param.cpp [hunk 2]
```diff
@@ -1222,7 +1222,7 @@ OPENMS_THREAD_CRITICAL(oms_log)
             }
             this->root_.insert(local_entry, prefix); //->setValue(it.getName(), local_entry.value, local_entry.description, local_entry.tags);
           }
-          else
+          else if(verbose)
```
Comment (timosachsenberg) [inline src/openms/source/DATASTRUCTURES/Param.cpp:1225]: so this is now silenced? might be a bit dangerous

File: src/openms_gui/include/OpenMS/VISUAL/DIALOGS/SwathTabWidget.h [hunk 3]
```diff
@@ -46,23 +47,53 @@ namespace Ui
...
+      Param swath_param_; ///< the global Swath parameters which will be passed to OpenSwathWorkflow.exe, once updated with parameters the Wizard holds separately
+      Param swath_param_wizard_; ///< small selection of important parameters which the user can directly change in the Wizard
     };
 
   }
-}
+} // ns OpenMS
+
+// this is required to allow Ui_SwathTabWidget (auto UIC'd from .ui) to have a InputFile member
+using InputFile = OpenMS::InputFile;
```
Comment (timosachsenberg) [inline src/openms_gui/include/OpenMS/VISUAL/DIALOGS/SwathTabWidget.h:97]: would it make sense to put them in an extra GUI or VISUAL namespace?

Comment (timosachsenberg) [reply src/openms_gui/include/OpenMS/VISUAL/DIALOGS/SwathTabWidget.h:97]: Ui ?

File: src/openms_gui/source/VISUAL/DIALOGS/SwathTabWidget.cpp [hunk 4]
```diff
@@ -62,19 +84,190 @@ namespace OpenMS
...
+        connect(ui->input_swath_windows, &InputFile::updatedCWD, this, &SwathTabWidget::broadcastNewCWD_);
     }
 
     SwathTabWidget::~SwathTabWidget()
     {
         delete ui;
     }
 
-    void SwathTabWidget::on_pushButton_clicked()
+    String infileToOSW(const String& infile)
```
Comment (timosachsenberg) [inline src/openms_gui/source/VISUAL/DIALOGS/SwathTabWidget.cpp:120]: needed as public member?

**Comment by timosachsenberg** [comment]
looks good but it seems some tests are failing.
Screenshots?
Is this already in a state that e.g. @Leon-Bichmann could test it?

File: src/openms/source/APPLICATIONS/TOPPBase.cpp (hunk 1 above)
Comment (cbielow) [reply src/openms/source/APPLICATIONS/TOPPBase.cpp:1604]: it was already .. see some lines above.

File: src/openms/source/DATASTRUCTURES/Param.cpp (hunk 2 above)
Comment (cbielow) [reply src/openms/source/DATASTRUCTURES/Param.cpp:1225]: if you do 'verbose == true' then its active again :)
But you mean for other use cases like `-write_ini + -ini` ?

Comment (timosachsenberg) [reply src/openms/source/DATASTRUCTURES/Param.cpp:1225]: yep. Is it active by default if one uses an old ini file in a new version?

File: src/openms_gui/include/OpenMS/VISUAL/DIALOGS/SwathTabWidget.h (hunk 3 above)
Comment (cbielow) [reply src/openms_gui/include/OpenMS/VISUAL/DIALOGS/SwathTabWidget.h:97]: the problem is the auto-generated header from the .ui file: it references an `InputFile* obj;`. 
We'd need to convince the UIC to generate `OpenMS::Inputfile* obj` or similar, but I did not bother to look for how to do that (yet)

File: src/openms_gui/source/VISUAL/DIALOGS/SwathTabWidget.cpp (hunk 4 above)
Comment (cbielow) [reply src/openms_gui/source/VISUAL/DIALOGS/SwathTabWidget.cpp:120]: no. Its only used in SwathWidget.cpp. No need to export this.

**Comment by cbielow** [comment]
feel free to test; not very exciting ATM though :)
image

**Comment by cbielow** [comment]
tests pass. Any further comments?

**Pull request merged**